### DISCUSSION TITLE
FINDING CONJUNCTIONS

### DISCUSSION THREAD

**Discussion by NorthBaandee** [skyfielders/python-skyfield#780] [Q&A]
I'd like to write a script that finds conjunctions or reasonably close approaches (say <15degrees) of comets and minor planets with naked eye stars or planets (ie anything brighter than abt +2).
Would I use find_minima() and separation_from() for this?

**Comment by brandon-rhodes** [comment]
There are enough objects brighter than +2 (dozens at least, I think? have you counted?) that taking every comet and running `find_minima()` between the comet and each object would be ruinously expensive. Probably you will want to pre-compute daily or weekly positions for all the +2 objects over the time period you are interested in. Then, for each comet, compute positions for the same dates and then run `separation_from()` between that series of _n_ positions and the _n_ positions of each +2 object, looking for dates when the separation falls below 15°.

That way, all those star and planet positions only get computed once.

If you needed to be really fancy, you could try concatenating all the +2 positions into a single vector, but that would be a lot of extra complexity and I'm not sure how much it would affect the runtime—maybe not enough to matter?

**Reply by NorthBaandee** [reply]
Actually +2 is probably too dim for what I want to achieve. I write an astronomy column for the local paper. The idea is to help people who probably can't tell Arcturus from Fomalhaut roughly where to look for things, so I can write stuff like "a hands width away from the bright red star in the north-east" etc

**Reply by brandon-rhodes** [reply]
Ah, understood, thank you for the explanation! I think you will want to use `separation_from()` and maybe `position_angle_of()` to tell folks whether to look up or down or left or right—here's a sample script that shows its use:

https://rhodesmill.org/skyfield/examples.html#at-what-angle-in-the-sky-is-the-crescent-moon

A simple loop over bright objects should hopefully let you focus in on the closest ones to the object you are trying to help folks find.

**Reply by NorthBaandee** [reply]
The main requests I get are for comets, Vesta ("Hey I heard there is one asteroid you can see with the naked eye! Can you tell us where it is,????") and Uranus.
BTW the url for hipparcos has expired certificates so I have downloaded it from Harvard.
